speck/tests/mod.rs

```diff
@@ -0,0 +1,168 @@
+//! Test vectors are from The Simon and Speck Families of Lightweight Block Ciphers (Appendix C)
+
+use cipher::{BlockDecrypt, BlockEncrypt, KeyInit};
+use hex_literal::hex;
+use speck::{
+    Speck128_128, Speck128_192, Speck128_256, Speck32_64, Speck48_72, Speck48_96, Speck64_128,
+    Speck64_96, Speck96_144, Speck96_96,
+};
+
+#[test]
+fn speck32_64() {
+    let key = hex!("1918111009080100");
+    let plaintext = hex!("6574694c");
+    let ciphertext = hex!("a86842f2");
+    let cipher = Speck32_64::new(&key.into());
```

Comment:
BTW it could be worth to write a macro, so you could define the tests like this:
```rust
new_tests!(
    { Speck32_64, speck32_64, hex!("1918111009080100"), hex!("6574694c"), hex!("a86842f2") }
    { Speck48_72, speck48_72, hex!("1211100a0908020100"), hex!("20796c6c6172"), hex!("c049a5385adc") }
    // ...
)
```